bin/problem.py

```diff
@@ -889,3 +908,94 @@ def process_testcase(testcase):
                     )
 
         return success
+
+    def determine_timelimit(problem):
+        ts_pair = problem.prepare_run()
+        if ts_pair == False:
+            return False
+        testcases, submissions = ts_pair
+
+        max_submission_len = max([len(x.name) for cat in submissions for x in submissions[cat]])
+
+        problem.settings.timelimit = float('inf')
+        problem.settings.timelimit_is_default = False
+        problem.settings.timeout = float('inf')
+
+        ok = True
+
+        def run_all(cur_verdicts, select):
+            nonlocal ok
+
+            cur_submissions = {v: submissions[v] for v in cur_verdicts}
```

Comment:
As discussed, please change `submissions` to just be a flat list and filter for submissions with `s.expected_verdicts = [v]`.